### ISSUE TITLE
Access generated code for ElementwiseKernel and ReductionKernel

### ISSUE THREAD

**Issue by Indivicivet** [cupy/cupy#6969]
### Description

The result of the experimental `cupyx.jit.rawkernel` decorator exposes a `.cached_code` parameter, which is useful for observing exactly what is being compiled, especially as a cupy beginner with CUDA experience.

It would be useful for this to be exposed also for ElementwiseKernel and ReductionKernel. There is the added upside that these could be used for quickly generating such "simple-case" kernels and then modifying them to have additional behaviour which cannot be captured with actual instances of these kernels (e.g. modified loop conditions).

I believe this could be implemented by additionally returning the variable `module_code` generated in `cupy/_core/_kernel.pyx`'s `_get_simple_elementwise_kernel() ` and saving it as a member variable in the ElementwiseKernel / ReductionKernel object, and making it accessible as e.g. `.cached_code` or `.module_code` or similar.

It would also be useful if the CUPY_FOR macro was either expanded inline or included as preamble in the resulting string.

```
// CArray
#define CUPY_FOR(i, n) \
    for (ptrdiff_t i = \
            static_cast<ptrdiff_t>(blockIdx.x) * blockDim.x + threadIdx.x; \
         i < (n); \
         i += static_cast<ptrdiff_t>(blockDim.x) * gridDim.x)
```

**Comment by kmaehashi** [comment]
Thanks for the feedback @Indivicivet! Adding `cached_code(s)` to these kernels sounds good to me. PRs welcomed. So far you can use `CUPY_CACHE_SAVE_CUDA_SOURCE` (see #1243) but attributes are much easier to access.

**Comment by kmaehashi** [comment]
Implemented via #7265
